lib_comfyui/queue_tracker.py

```diff
@@ -0,0 +1,98 @@
+import multiprocessing
+from lib_comfyui import ipc
+
+
+class PromptQueueTracker:
+    done_event = multiprocessing.Event()
+    tracked_id = None
+    queue_instance = None
+    server_instance = None
+
+    @ipc.confine_to('comfyui')
+    @staticmethod
+    def wait_for_last_put():
+        if not PromptQueueTracker.tracked_id_present():
+            return
+
+        PromptQueueTracker.done_event.wait()
+        PromptQueueTracker.done_event.clear()
+
+    @staticmethod
+    def tracked_id_present():
+        with PromptQueueTracker.queue_instance.mutex:
+            for _, v in PromptQueueTracker.queue_instance.currently_running.items():
+                if abs(v[0]) == PromptQueueTracker.tracked_id:
+                    return True
+            for x in PromptQueueTracker.queue_instance.queue:
+                if abs(x[0]) == PromptQueueTracker.tracked_id:
+                    return True
+            return False
+
+
+    @staticmethod
+    def update_tracked_id():

```

Comment:
I know the code isn't complete, but I have seen this and I don't want to forget.

should this function be `@confine_to('comfyui')`?

Reply:
And what about `tracked_id_present()` and maybe other functions of the class here